iothub/service/src/Jobs/ScheduledJobsClient.cs

```diff
@@ -0,0 +1,378 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Scheduled Jobs management.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-jobs"/>.
+    public class ScheduledJobsClient
+    {
+        private string _hostName;
+        private IotHubConnectionProperties _credentialProvider;
+        private HttpClient _httpClient;
+        private HttpRequestMessageFactory _httpRequestMessageFactory;
+
+        private const string JobsUriFormat = "/jobs/v2/{0}";
+        private const string JobsQueryFormat = "/jobs/v2/query";
+        private const string CancelJobUriFormat = "/jobs/v2/{0}/cancel";
+
+        private const string ContinuationTokenHeader = "x-ms-continuation";
+        private const string PageSizeHeader = "x-ms-max-item-count";
+
+        private static readonly TimeSpan s_defaultOperationTimeout = TimeSpan.FromSeconds(100);
+
+        /// <summary>
+        /// Creates scheduled jobs client, provided for unit testing purposes only.
+        /// </summary>
+        public ScheduledJobsClient()
+        {
+        }
+
+        internal ScheduledJobsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _credentialProvider = credentialProvider;
+            _hostName = hostName;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Explicitly open the ScheduledJobsClient instance.
```

Comment:
Avoid hard coding the client name.

```suggestion
        /// Explicitly open the client instance.
```